repo/blob/azure/azure_storage.go

```diff
@@ -142,6 +151,22 @@ func (az *azStorage) PutBlob(ctx context.Context, b blob.ID, data blob.Bytes, op
 		return translateError(err)
 	}
 
+	if opts.RetentionPeriod != 0 {
+		// it will fail if the retentionPeriod set by the user is lower than that on the policy.
+		retainUntilDate := clock.Now().Add(opts.RetentionPeriod).UTC()
+		mode := azblobblob.ImmutabilityPolicySetting(opts.RetentionMode)
```

Comment:
It will work. The user just needs to make sure they set the `RetentionPeriod` to the same or greater than that of the policy. Just need to add a note like https://kopia.io/docs/advanced/ransomware-protection/#an-additional-note-about-lifecycle-management-vs-retention-time saying they must set it to the same or higher; it will fail early. Can also return a more specific error message when this error occurs so they know the reason (e.g. when it fails for this tell them the retention period is too low vs the policy's default)